main_test.go

```diff
@@ -0,0 +1,282 @@
+package main
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/vmware/cluster-api-upgrade-tool/pkg/upgrade"
+)
+
+func TestUpgradeScenario(t *testing.T) {
+	// spin up management cluster
+	setupManagementCluster(t)
+
+	clusterName := "my-cluster"
```

Comment:
Should we use a common prefix and generate a random suffix for this & the namespace?